cstrml/staking/src/lib.rs

```diff
@@ -1676,37 +1799,39 @@ impl<T: Trait> Module<T> {
             let points = CurrentEraPointsEarned::take();
             let validators = Self::current_elected();
             let total_authoring_payout = Self::authoring_rewards_in_era();
-            let mut total_imbalance = <PositiveImbalanceOf<T>>::zero();
-            let to_num =
-                |b: BalanceOf<T>| <T::CurrencyToVote as Convert<BalanceOf<T>, u64>>::convert(b);
-
+            // let mut total_imbalance = <PositiveImbalanceOf<T>>::zero();
+            let current_era = Self::current_era().unwrap_or(0);
             // 1. Block authoring payout
             for (v, p) in validators.iter().zip(points.individual.into_iter()) {
                 if p != 0 {
                     let authoring_reward =
                         Perbill::from_rational_approximation(p, points.total) * total_authoring_payout;
-                    total_imbalance.subsume(Self::reward_author(v, authoring_reward));
+                    <ErasAuthoringPayout<T>>::insert(&current_era, v, authoring_reward);
                 }
             }
 
             // 2. Staking payout
-            let current_era = Self::current_era().unwrap_or(0);
             let total_staking_payout = Self::staking_rewards_in_era(current_era);
-            let total_stakes = Self::total_stakes();
-            <Stakers<T>>::iter().for_each(|(v, e)| {
-                let staking_reward = Perbill::from_rational_approximation(to_num(e.total), to_num(total_stakes)) * total_staking_payout;
-                total_imbalance.subsume(Self::reward_validator(&v, staking_reward));
-            });
+            <ErasStakingPayout<T>>::insert(&current_era, total_staking_payout);
 
-            // 3. Deposit reward event
-            Self::deposit_event(RawEvent::Reward(total_authoring_payout, total_staking_payout));
+            // 3. Deposit era reward event
+            Self::deposit_event(RawEvent::EraReward(current_era, total_authoring_payout, total_staking_payout));
 
-            T::Reward::on_unbalanced(total_imbalance);
+            // TODO: enable treasury and might bring this back
+            // T::Reward::on_unbalanced(total_imbalance);
```

Comment:
Curently we don't use this reward at all(even before this PR). Substrate uses treasury runtime. We might enable this back when we integrate treasury into the staking.

Reply:
cool, make sense🙏🏻